src/Xiaomi_Scale.py

```diff
@@ -176,16 +149,21 @@ def _publish(self, weight, unit, mitdatetime, hasImpedance, miimpedance):
         message += ',"TimeStamp":"' + mitdatetime + '"'
         message += '}'
         try:
-          self.mqtt_client.publish(MQTT_PREFIX + '/' + user + '/weight', message, qos=1, retain=True)
-          sys.stdout.write('Sent data to topic %s: %s' % (MQTT_PREFIX + '/' + user + '/weight', message + '\n'))
+            sys.stdout.write('Sent data to topic %s: %s' % (MQTT_PREFIX + '/' + user + '/weight', message + '\n'))
+            publish.single(
+                MQTT_PREFIX + '/' + user + '/weight',
+                message,
+                qos=1,
+                retain=True,
+                hostname=MQTT_HOST,
+                port=MQTT_PORT,
+                auth={'username':MQTT_USERNAME, 'password':MQTT_PASSWORD}
+            )
         except:
-          sys.stdout.write('Could not publish to MQTT, Disconnecting...\n')
-          MQTT_CONNECTED = False
-          self.mqtt_client.disconnect()
-          pass
+          sys.stdout.write('Could not publish to MQTT\n')
```

Comment:
instead of a simple stdout I think we need to display the actual error message to give the user a hint. That's probably true for all stdout when an exception occurs